applications/nrf_desktop/doc/smp_custom.rst

```diff
@@ -0,0 +1,81 @@
+
+Configuration
+*************
+
+Set the :ref:`CONFIG_DESKTOP_DFU_MCUMGR_ENABLE <config_desktop_app_options>` Kconfig option to enable the module.
+
+The module enables the following configurations:
+
+* :kconfig:option:`CONFIG_MCUMGR` - This option enables MCUmgr support, which is required for the DFU process.
+  For details, see :ref:`zephyr:mcu_mgr` in the Zephyr documentation.
+* MCUmgr groups:
+  * :kconfig:option:`CONFIG_MCUMGR_GRP_IMG` - This option enables MCUmgr image management handlers, which are required for the DFU process.
+    For details, see :ref:`zephyr:device_mgmt` in the Zephyr documentation.
+  * :kconfig:option:`CONFIG_MCUMGR_GRP_OS` - This option enables MCUmgr OS management handlers, which are required for the DFU process.
+    For details, see :ref:`zephyr:device_mgmt` in the Zephyr documentation.
+* :kconfig:option:`CONFIG_MCUMGR_MGMT_NOTIFICATION_HOOKS` - This option enables MCUmgr notification hook support, which allows this module to listen for a MCUmgr event.
+  For details, see :ref:`zephyr:mcumgr_callbacks` in the Zephyr documentation.
+  MCUmgr notifications hooks:
+    * :kconfig:option:`CONFIG_MCUMGR_GRP_IMG_UPLOAD_CHECK_HOOK` - This option enables MCUmgr upload check hook, which sends image upload requests to the registered callbacks.
+    * :kconfig:option:`CONFIG_MCUMGR_GRP_IMG_STATUS_HOOKS` - This option enables MCUmgr image status hooks, which report the DFU status to the registered callbacks.
+    * :kconfig:option:`CONFIG_MCUMGR_GRP_OS_RESET_HOOK` - This option enables MCUmgr OS reset hook, which sends reset requests to the registered callbacks.
+* MCUmgr dependencies:
+  * :kconfig:option:`CONFIG_NET_BUF`.
+  * :kconfig:option:`CONFIG_ZCBOR`.
+  * :kconfig:option:`CONFIG_CRC`.
+  * :kconfig:option:`CONFIG_IMG_MANAGER`.
+  * :kconfig:option:`CONFIG_STREAM_FLASH`.
+  * :kconfig:option:`CONFIG_FLASH_MAP`.
+  * :kconfig:option:`CONFIG_FLASH`.
+
+To use the module, you must also enable the :kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` option.
+The DFU over Simple Management Protocol in Zephyr is supported only with the MCUboot bootloader.
+
+You must also enable the preferred DFU transport option for the MCUmgr module (e.g. :kconfig:option:`CONFIG_MCUMGR_TRANSPORT_BT` option).
+With the :kconfig:option:`CONFIG_MCUMGR_TRANSPORT_BT` option enabled, the module enables the :kconfig:option:`CONFIG_CAF_BLE_SMP_TRANSFER_EVENTS` event.
+The event is used to lower Bluetooth connection latency during the DFU process.
+
+The DFU module leverages the :ref:`nrf_desktop_dfu_lock` to synchronize flash access with other DFU methods (e.g. Config Channel DFU).
+Set the :ref:`CONFIG_DESKTOP_DFU_LOCK <config_desktop_app_options>` Kconfig option to enable this feature.
+Make sure that the DFU lock utility is enabled if your nRF Desktop application configuration uses multiple DFU transports.
+
+The custom |smp| implementation is currently marked as experimental because it uses internal MCUmgr API.
```

Comment:
This could be made as a note and moved to introductory section (above the Module events section) because I believe this is something the user should know at the start.